syntax = "proto3";

package rps;

enum HandResult {
  P1_WON = 0;
  P2_WON = 1;
  DRAW = 2;
}

enum Decision {
  NONE = 0;
  ROCK = 1;
  PAPER = 2;
  SCISSOR = 3;
}

message EnvConfig {}

message GameState {
  int32 p1_score = 1;
  int32 p2_score = 2;
}

message GameStateDelta {
  HandResult result = 1;

  Decision p1_decision = 2;
  Decision p2_decision = 3;
}

message ActorAction {  
  Decision decision = 1;
}

message JudgeView {}

message JudgeAction {
  HandResult decision = 1;
}
